lib/ansible/modules/cloud/vmware/vmware_dvswitch_netflow.py

```diff
@@ -0,0 +1,273 @@
+        self.dvs = find_dvs_by_name(self.content, self.switch_name)
+        if self.dvs is None:
+            self.module.fail_json(msg="Failed to find DVS %s" % self.switch_name)
+
+    def ensure(self):
+        """Manage NetFlow configuration"""
+        changed = False
+        results = dict(changed=changed)
+        results['dvswitch'] = self.switch_name
+        changed_list = []
+
+        config_spec = vim.dvs.VmwareDistributedVirtualSwitch.ConfigSpec()
+        # Use the same version in the new spec; The version will be increased by one by the API automatically
+        config_spec.configVersion = self.dvs.config.configVersion
+
+        # Check NetFlow
+        results['collector_ip'] = self.netflow_collector_ip
+        results['collector_port'] = self.netflow_collector_port
+        results['observation_domain'] = self.netflow_observation_domain
+        # results['switch_ip'] = self.netflow_switch_ip
+        results['active_flow_export_timeout'] = self.netflow_active_flow_export
+        results['idle_flow_export_timeout'] = self.netflow_idle_flow_export
+        results['sampling_rate'] = self.netflow_sampling_rate
+        results['process_internal_flows_only'] = self.netflow_process_internal_flows
+        netflow_config = self.dvs.config.ipfixConfig
+        netflow_config_spec = vim.dvs.VmwareDistributedVirtualSwitch.IpfixConfig()
+        netflow_config_spec.collectorIpAddress = self.netflow_collector_ip
+        netflow_config_spec.collectorPort = self.netflow_collector_port
+        netflow_config_spec.observationDomainId = self.netflow_observation_domain
+        netflow_config_spec.activeFlowTimeout = self.netflow_active_flow_export
+        netflow_config_spec.idleFlowTimeout = self.netflow_idle_flow_export
+        netflow_config_spec.samplingRate = self.netflow_sampling_rate
+        netflow_config_spec.internalFlowsOnly = self.netflow_process_internal_flows
+        # the collectorIpAddress can be '' or None
+        if not (self.netflow_collector_ip == ''
+                and (netflow_config.collectorIpAddress == '' or netflow_config.collectorIpAddress is None)):
+            if netflow_config.collectorIpAddress != self.netflow_collector_ip:
+                changed = True
+                changed_list.append("collector IP address")
+                results['collector_ip_previous'] = netflow_config.collectorIpAddress
+        if netflow_config.collectorPort != self.netflow_collector_port:
+            changed = True
+            changed_list.append("collector port")
+            results['collector_port_previous'] = netflow_config.collectorPort
+        if netflow_config.observationDomainId != self.netflow_observation_domain:
+            changed = True
+            changed_list.append("observation domain ID")
+            results['observation_domain_previous'] = netflow_config.observationDomainId
+        # looks like switch IP can't be set via SOAP API!?
```

Comment:
This can be set, but not in the netflow/ipfix, config. It can be set on the dvSwitch itself using its configSpec:
https://code.vmware.com/apis/358#/doc/vim.DistributedVirtualSwitch.ConfigSpec.html#switchIpAddress

Reply:
ok. Thanks. I will test this when I'm back from vacation.